core/src/main/java/io/cucumber/core/runner/HookTestStep.java

```diff
@@ -8,6 +8,7 @@
 
     private final HookType hookType;
     private final HookDefinitionMatch definitionMatch;
+    private io.cucumber.plugin.event.TestStep relatedTestStep;
```

Comment:
Instead of making test steps mutable consider passing the current step along with the `TestCaseState` and wrapping it in a `Step` object in the same way the `Scenario` object wraps the `TestCaseState`. This ensures the information passed to hooks is consistently.  It also ensures that an illegal state exception can be thrown when `Step` object is used outside of an active step hook.

Reply:
Does this mean I need to create a new class **io.cucumber.java.Step** for this purpose?

Reply:
Yes please.